[jsx-no-script-url]: use shared settings to specify link components (#3673)
jsx-eslint/eslint-plugin-react: docs/rules/jsx-no-script-url.md

```

Allows you to indicate a specific list of properties used by a custom component to be checked.
This will override anything passed to `linkComponents` setting.

> NOTE: This rule now takes into account `linkComponents` setting and it should be used as primary source of link components.
> The rule still allows passing link components as rule option, but it is meant only as backwards-compatibility feature.
> New setups should only use `linkComponents` setting.

Reviewer: ```suggestion
NOTE: This rule now takes into account the `linkComponents` setting, which should be used as the sole source of truth for link components.
The rule still allows passing link components as rule option for backwards compatibility, but this option is deprecated.
```
